proxy/src/http.rs

```diff
@@ -0,0 +1,54 @@
+//! HTTP API delivering JSON over `RESTish` endpoints.
+
+use librad::paths;
+use std::convert::Infallible;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+use warp::{path, Filter};
+
+use crate::registry;
+
+mod error;
+mod notification;
+mod project;
+
+/// Main entry point for HTTP API.
+pub async fn run(librad_paths: paths::Paths, reg: registry::Registry) {
+    let subscriptions = crate::notification::Subscriptions::default();
+
+    let api = path("v1").and(
```

Comment:
This should be `path("api").and(path("v1"))`, that is, `/api/v1`, in keeping with typical API [(1)][1][(2)][2] namespacing.

[1]: https://docs.joinmastodon.org/methods/apps/
[2]: https://developer.twitter.com/en/docs/accounts-and-users/create-manage-lists/api-reference/get-lists-list

Reply:
I don't think there is a general convention, best-practice here. Unless I missed an obvious standard body or something similar.

Reply:
Having `api` distinctly in the path makes sense if the server offers other endpoints as well. For our case the proxy is literally the API itself.

Reply:
Well, how about telemetry or debugging/metrics endpoints? Won't we have those somewhere in there?

Reply:
Hypothetically even those metrics and debugging endpoints would be namespaced under `/v1/`, as `/v1/stats` in anticipation of a potential `/v2/stats` or `/v2/stats_but_different`.

Mastodon and others use a `/api/v1/` path because they expose other non-API HTTP endpoints. Twitter puts `api` in the subdomain rather than the path because the responding server only handles an API, which is more like our use-case as the proxy *is* the API.